static/app/utils/useLocalStorageState.ts

```diff
@@ -0,0 +1,157 @@
+import {useCallback, useLayoutEffect, useState} from 'react';
+
+const SUPPORTS_QUEUE_MICROTASK = window && 'queueMicrotask' in window;
+const SUPPORTS_LOCAL_STORAGE = window && 'localStorage' in window;
+
+function scheduleMicroTask(callback: () => void) {
```

Comment:
I like the extremely defensive programming here. It makes me wonder though, how much are we expected to support old browsers here? `queueMicrotask` seems new enough for this guard to make sense, but checking for `setTimeout` isn't something I'd normally do. So I guess I'm asking if I need to be this defensive when I'm adding code. And a related question: how much are we polyfilling, if at all?

If this amount of guarding is necessary to use queueMicrotask, I'd love for this to be its own common util 🙂

Reply:
Oh yeah, that is definitely a mistake and we should not check that. I'll just check for window in case this runs on the server at some point, but that was not intended...